Xamarin.Essentials/Contacts/Contact.ios.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Contacts;
+using ContactsUI;
+using Foundation;
+using UIKit;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Contact
+    {
+        internal static Action<PhoneContact> CallBack { get; set; }
+
+        static Task<PhoneContact> PlatformPickContactAsync()
+        {
+            var uiView = Platform.GetCurrentViewController();
+            if (uiView == null)
+                throw new ArgumentNullException($"The View Controller can't be null.");
+
+            using var picker = new CNContactPickerViewController
+            {
+                Delegate = new ContactPickerDelegate()
+            };
+
+            uiView.PresentViewController(picker, true, null);
+            var source = new TaskCompletionSource<PhoneContact>();
+            try
+            {
+                CallBack = (phoneContact) =>
+                {
+                    var tcs = Interlocked.Exchange(ref source, null);
+                    tcs?.SetResult(phoneContact);
+                };
+            }
+            catch (Exception ex)
+            {
+                source.SetException(ex);
+            }
+            return source.Task;
+        }
+
+        internal static PhoneContact GetContact(CNContact contact)
+        {
+            if (contact == null)
+                return default;
+
+            try
+            {
+                var contactType = ToPhoneContact(contact.ContactType);
+                var phones = new Dictionary<string, ContactType>();
+
+                foreach (var item in contact.PhoneNumbers)
+                    phones.Add(item?.Value?.StringValue, contactType);
```

Comment:
The same contact type is used for all numbers and is based on the mapping from `CNContactType`. After testing this code on iOS, the behaviour of the API is to return `CNContactType.Person` for any contact that has a given name and family set. It returns `CNContactType.Organization` only for contacts where these values are empty, but the company name is set.

Instead, I think the correct approach would be to add a new mapping function that maps the `item.Label` to `ContactType`, based on constants defined in `CNLabelPhoneNumberKey`, such as `CNLabelPhoneNumberKey.Mobile` or `CNLabelPhoneNumberKey.Main`, or raw values such as `_$!<Work>!$_` (which is the label for work number, but which is not assigned a const in the API). This would be more aligned with the UWP `GetPhoneContactType` method.

Perhaps a new enum of type `PhoneType` could be considered to distinguish between mobile and other? But I totally see the point of keeping the API small and simple.